src/mlir/generator.cc

```diff
@@ -395,31 +409,86 @@ namespace mlir::verona
 
   Value MLIRGenerator::ConstantString(StringRef str, StringRef name)
   {
-    // Use auto-generated name if none provided
-    static size_t incr = 0;
-    std::string nameStr;
+    // Use contents as name if none provided
     if (name.empty())
-      nameStr = "_string" + std::to_string(incr++);
-    else
-      nameStr = name.str();
-
-    // In LLVM, strings are arrays of i8 elements
-    auto i8 = builder.getIntegerType(8);
-    auto strTy = ArrayType::get(i8, str.size());
-    auto strAttr = builder.getStringAttr(str);
-
-    // In LLVM, constant strings are global objects
-    auto moduleBuilder = OpBuilder(*module);
-    auto global = moduleBuilder.create<LLVM::GlobalOp>(
-      builder.getUnknownLoc(),
-      strTy,
-      /*isConstant=*/true,
-      LLVM::Linkage::Private,
-      nameStr,
-      strAttr);
-    module->push_back(global);
+      name = str;
+
+    // Avoid redefinition
+    auto global = module->lookupSymbol<LLVM::GlobalOp>(name);
+    if (!global)
+    {
+      // In LLVM, strings are arrays of i8 elements
+      auto i8 = builder.getIntegerType(8);
+      auto strTy = ArrayType::get(i8, str.size());
+      auto strAttr = builder.getStringAttr(str);
+
+      // In LLVM, constant strings are global objects
+      auto moduleBuilder = OpBuilder(*module);
+      global = moduleBuilder.create<LLVM::GlobalOp>(
+        builder.getUnknownLoc(),
+        strTy,
+        /*isConstant=*/true,
+        LLVM::Linkage::Private,
+        name,
+        strAttr);
+      module->push_back(global);
+    }
 
     // But their addresses are a local operation
     return builder.create<LLVM::AddressOfOp>(builder.getUnknownLoc(), global);
   }
+
+  Value MLIRGenerator::Arithmetic(Location loc, StringRef name, Value ops)
+  {
+    // FIXME: We already converted U32 to i32 so this "works". But we need
+    // to make sure we want that conversion as early as it is, and if not,
+    // we need to implement this as a standard select and convert that
+    // later. However, that would only work if U32 has a method named "+",
+    // or if we declare it on the fly and then clean up when we remove the
+    // call.
+    auto numOps = arithmetic[name];
```

Comment:
Use `arithmetic.at(name)` instead. This will swallow invalid names and use `0` as the arity.

Reply:
Wait, this doesn't look right.
https://www.cplusplus.com/reference/map/map/at/ says:
```
Returns a reference to the mapped value of the element identified with key k.

If k does not match the key of any element in the container, the function throws an out_of_range exception.
```
But also that the operator `[]` creates an empty element if none exists, which is not what I want there either. 

I'm changing it to the common `find` pattern, which is what I should have done in the first place. :)
```
auto it = ...find(name);
assert(it != ...end());
auto numOps = it->second;
```

Reply:
`.at` is what you want here.  You then don't need the `assert`, because the uncaught exception will end the program.  If you compile without exceptions, [this aborts if it doesn't find the value](https://github.com/llvm/llvm-project/blob/bfbd73f87d7b2c1140b43192cea6b7fbbfc35cd0/libcxx/include/stdexcept#L270).

Reply:
exception/aborting is worse than asserts because it provides no context. Though, now we're using `StringSwitch` and this isn't relevant any longer.